ui/shared/utils/constants.js

```diff
@@ -1370,7 +1374,7 @@ export const predictionFieldValue = (
 
   return indicatorMap[value[0]] || indicatorMap[value]
 }
-export const predictorColorRanges = thresholds => (
+export const predictorColorRanges = (thresholds, requiresCitation = true) => (
```

Comment:
I'm not sure that defaulting this to true is the behavior we want - is there any case where we are calling this function without explicitly passing `requiresCitation`?

Reply:
We call this function when `requiresCitation` is sometimes `undefined`. I only set the value of `requiresCitation` for those predictors in `ORDERED_PREDICTOR_FIELDS` when the value is`false`, and for the rest of the predictors `requiresCitation` is `undefined`. The fn `predictorColorRanges` is called twice, and in both calls I contemplated defaulting `requiresCitation` to `true` if it is `undefined` before the call, but decided to do it here so defaulting only happens once.

Another option is just to set `requiresCitation: true` in each field in `ORDERED_PREDICTOR_FIELDS` and have no default logic, and make `requiresCitation` a required prop on `Prediction`

Reply:
I think I am a little confused why we would want to be showing the citation unless we were sure it was a field where the citation was appropriate. So if the function is called twice, shouldn't we explicitly be setting `requiresCitation` in both of those calls, to make sure the behavior is correct in both places? Also, as discussed on the previous review, we really want the citation to be a whitelist and not a blacklist. So setting it to false in a few places opens us up to the likelihood that as we add new predictions we do not remember to set the citation field to false. Since there is a finite list of scores in the paper that we want to cite, I think it is better practice to explicitly set the citation to true in those cases, and ave the default behavior be no citation

Reply:
ahh I see how setting `requiresCitation` to `false` only for some predictors is still just blacklisting them. If I did the opposite and just added `requiresCitation: true` to those fields that we want citations for, that would be whitelisting?